manifests/statefulset_test.go

```diff
@@ -0,0 +1,395 @@
+package manifests_test
+
+import (
+	"reflect"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openlyinc/pointy"
+
+	"github.com/freeipa/freeipa-operator/api/v1alpha1"
+	manifests "github.com/freeipa/freeipa-operator/manifests"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func assertStringListsEqual(data1 []string, data2 []string) {
+	eq := reflect.DeepEqual(data1, data2)
+	Expect(eq).Should(BeTrue())
+}
+
+func assertEnvVarEqual(data1 *corev1.EnvVar, data2 *corev1.EnvVar) {
+	if data1 == nil && data2 == nil {
+		return
+	}
+	Expect(data1 != nil && data2 != nil).Should(BeTrue())
+	eq := reflect.DeepEqual(data1, data2)
+	Expect(eq).Should(BeTrue())
+}
+
+func assertContainerPortEqual(data1 *corev1.ContainerPort, data2 *corev1.ContainerPort) {
+	if data1 == nil && data2 == nil {
+		return
+	}
+	Expect(data1 != nil && data2 != nil).Should(BeTrue())
+	eq := reflect.DeepEqual(data1, data2)
+	Expect(eq).Should(BeTrue())
+}
+
+func assertVolumeMountEqual(data1 *corev1.VolumeMount, data2 *corev1.VolumeMount) {
+	if data1 == nil && data2 == nil {
+		return
+	}
+	Expect(data1 != nil && data2 != nil).Should(BeTrue())
+	eq := reflect.DeepEqual(data1, data2)
+	Expect(eq).Should(BeTrue())
+}
+
+func assertVolumeEqual(data1 *corev1.Volume, data2 *corev1.Volume) {
+	if data1 == nil && data2 == nil {
+		return
+	}
+	Expect(data1 != nil && data2 != nil).Should(BeTrue())
+	eq := reflect.DeepEqual(data1, data2)
+	Expect(eq).Should(BeTrue())
+}
+
+func assertStringStringMapsEqual(data1 map[string]string, data2 map[string]string) {
+	// https://stackoverflow.com/questions/18208394/how-to-test-the-equivalence-of-maps-in-golang
+	eq := reflect.DeepEqual(data1, data2)
+	Expect(eq).Should(BeTrue())
+}
+
+func assertPodSecurityContext(data *corev1.SecurityContext) {
+	Expect(data).ShouldNot(BeNil())
+	Expect(data.Privileged).ShouldNot(BeNil())
+	Expect(*data.Privileged).Should(BeFalse())
+}
+
+var _ = Describe("LOCAL:Statefulset tests", func() {
+
+	Describe("MainStatefulsetForIDM", func() {
+		// GIVEN
+		Context("an IDM with PVC information", func() {
+			var idm = v1alpha1.IDM{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+					Labels: map[string]string{
+						"app": "freeipa",
+					},
+				},
+				Spec: v1alpha1.IDMSpec{
+					Realm:          "TEST.COM",
```

Comment:
We should use IPA.TEST, as it has a special TLD for testing purposes. See https://www.rfc-editor.org/rfc/rfc2606.html#section-2 test.com is someone else's domain and we shouldn't test against it.